packages/starlight/components/SidebarSublist.astro

```diff
@@ -21,7 +21,7 @@ interface Props {
             {entry.label}
           </a>
         ) : (
-          <details open>
+          <details open={!entry.collapsed}>
```

Comment:
I realised something while walking the dog this morning: what happens when you visit a page inside a collapsed section? And what would people expect?

In Vitepress or Docusaurus, the `collapsed` setting applies on initial load but as you expand sections and navigate pages, the sidebar stays in place, keeping its state. In Starlight, when you navigate to a new page, you’ll always return the server-rendered sidebar state.

I think we’d probably want to have the current page link visible on load, so maybe we need to account for that when setting the `open` attribute?

```suggestion
          <details
            open={entry.entries.flat(Infinity).some(i => i.isCurrent) || !entry.collapsed}
          >
```

Reply:
Oh yeah! In fact my naive `flat` version probably would not have worked at all now I think about it — nice one.